app/admin.js

```diff
@@ -7,7 +7,13 @@ const ArgumentParser = require('argparse').ArgumentParser;
 const pjson = require('../package.json');
 const fs = require('fs');
 const crypto = require('crypto');
-
+const Q = require('q');
```

Comment:
Q is pretty ancient, can you replace this with bluebird, or just vanilla promises if you can get away with it?